woocommerce/woocommerce
# Template API: Store edited product in context

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR closes #40233 and implements the following:

- `editedProduct` is available in `BlockContextProvider`
- `woocommerce/conditional`'s `mustMatch` attribute usage and implementation is updated to use the context (this is the first step in introducing broader conditional support in both `woocommerce/conditional` and the support for the conditional rendering of blocks



### How to test the changes in this Pull Request:



Manual testing:

1. Enable the **Product Block Editor** under **WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**.
2. Go to **Products** > **Add New**
3. Verify there are no regressions in the product editor. This PR does not introduce any functional changes. Some specific things to look for that would more likely be affected by these changes:
    - Tabs, including the currently selected state
    - The variations fields
    - Usage of the conditional block, such as on the inventory tab



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
